### PR TITLE
Used type_codes.inc in Visitor and BaseVisitor class

### PR DESCRIPTION
**Pull request by certik**
This finishes #539.

### CODE REVIEW

**Comment by isuruf** [comment]
@certik, since we need to exclude RealMPFR and ComplexMPC in some cases, how about,
```C++
#if defined(HAVE_SYMENGINE_MPFR) || defined(SYMENGINE_INCLUDE_ALL)
SYMENGINE_ENUM(REAL_MPFR, RealMPFR)
#endif
#if defined(HAVE_SYMENGINE_MPC) || defined(SYMENGINE_INCLUDE_ALL)
SYMENGINE_ENUM(COMPLEX_MPC, ComplexMPC)
#endif
```
and use `SYMENGINE_INCLUDE_ALL` where needed

**Comment by certik** [comment]
@isuruf where would you use `SYMENGINE_INCLUDE_ALL`? In all cases so far it seems we need to be able to include one or the other separately. In `basic.h` and in `cwrapper.h` we could use it, but the current code there is simpler. So I don't see how it would make things easier, or am I missing something?

**Comment by abinashmeher999** [comment]
@isuruf Wouldn't that make `TypeID_Count` inconsistent? If that is permissible we can do that too.

**Comment by isuruf** [comment]
Check out https://github.com/certik/symengine/compare/pr-539-fix...isuruf:types?expand=1 and see if it's okay

**Comment by certik** [comment]
I see. Yes, send a PR please against my branch.

**Comment by certik** [comment]
Thanks @isuruf this looks very nice. You can also notice that now when we automated this, it discovered a missing `accept` for FunctionWrapper.

File: symengine/type_codes.inc [hunk 1]
```diff
@@ -1,57 +1,56 @@
...
+SYMENGINE_ENUM(KRONECKERDELTA, KroneckerDelta)
+SYMENGINE_ENUM(LEVICIVITA, LeviCivita)
+SYMENGINE_ENUM(GAMMA, Gamma)
+SYMENGINE_ENUM(LOWERGAMMA, LowerGamma)
+SYMENGINE_ENUM(UPPERGAMMA, UpperGamma)
+SYMENGINE_ENUM(FUNCTIONSYMBOL, FunctionSymbol)
+SYMENGINE_ENUM(FUNCTIONWRAPPER, FunctionWrapper)
+SYMENGINE_ENUM(DERIVATIVE, Derivative)
+SYMENGINE_ENUM(SUBS, Subs)
+SYMENGINE_ENUM(ABS, Abs)
```
Comment (abinashmeher999) [inline symengine/type_codes.inc]: Ah I see. Removing `TypeID_Count` works too.

**Comment by certik** [comment]
Anyway, this passes all tests on Travis and simplifies the code a lot, so I think we can merge it.

**Pull request merged**